qtdialogs/DlgSetupManager.py

```diff
@@ -719,18 +593,40 @@ def unlockWallet(self, walletId):
          LOGINFO(f"Could not find wallet path for walletId: {walletId}")
          return
 
-      def onSuccess():
-         """Handle successful unlock - refresh wallet list."""
-         LOGINFO(f"Wallet {walletId} unlocked successfully")
-         self.loadWalletList()
+      try:
+         unlockDlg = UnlockWalletHandler(walletId, self.tr('Unlock Wallet'), self)
+
+         def onUnlockComplete(replyObj):
+            """Handle unlock control header reply."""
+            try:
+               if replyObj.success:
+                  LOGINFO(f"Wallet {walletId} unlocked successfully")
+                  TheSignalExecution.executeMethod(unlockDlg.accept)
```

Comment:
I don't understand the aggressive use of executeMethod. Just fire the entire callback from executeMethod if every possibly branch of execution has to run in the Qt thread anyways.